__tests__/app.test.js

```diff
@@ -8,82 +8,119 @@ const endpoints = require("../endpoints.json");
     it("should respond with a status code of 200 when user enters existing article_id", () => {
         return request(app)
-        .get("/api/articles/2")
-        .expect(200);
+            .get("/api/articles/2")
+            .expect(200);
     });
-    it("should respond with a single correctly formatted article object on a key of article", () => {
+    it("responds with a single object with all required properties on a key of article", () => {
         return request(app)
-        .get("/api/articles/3")
-        .then(({body}) => {
-            expect(body.article).toEqual(expect.objectContaining({
-                author: expect.any(String),
-                title: expect.any(String),
-                article_id: 3,
-                body: expect.any(String),
-                topic: expect.any(String),
-                created_at: expect.any(String),
-                votes: expect.any(Number),
-                article_img_url: expect.any(String)
-            }));
-        })
+            .get("/api/articles/3")
+            .then(({body}) => {
+                expect(body.article).toEqual(expect.objectContaining({
+                    author: expect.any(String),
+                    title: expect.any(String),
+                    article_id: 3,
+                    body: expect.any(String),
+                    topic: expect.any(String),
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    article_img_url: expect.any(String)
+                }));
+            })
     });
     it("should respond with a status code of 400: Bad request when user enters invalid input for article_id", () => {
         return request(app)
-        .get("/api/articles/nonsense")
-        .expect(400)
-        .then(({body}) => {
-            expect(body.msg).toBe("Bad request");
-        });
+            .get("/api/articles/nonsense")
+            .expect(400)
+            .then(({body}) => {
+                expect(body.msg).toBe("Bad request");
+            });
     });
     it("should respond with a status code of 404: Not found when user enters valid but non-existing input for article_id", () => {
         return request(app)
-        .get("/api/articles/9900")
-        .expect(404)
-        .then(({body}) => {
-            expect(body.msg).toBe("Not found");
-        });
+            .get("/api/articles/9900")
+            .expect(404)
+            .then(({body}) => {
+                expect(body.msg).toBe("Not found");
+            });
     });
-});
\ No newline at end of file
+});
+
+describe("GET /api/articles", () => {
+    it("responds with a status code of 200 and an array of all articles with required properties sorted in descending order by created_at", () => {
+        return request(app)
+            .get("/api/articles")
+            .expect(200)
+            .then(({body}) => {
+                expect(Array.isArray(body.articles)).toBe(true);
```

Comment:
I really like the array and length checks before you start looping to prevent any false positives.